cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -2099,7 +1770,7 @@ public boolean isExposedWithRoute() {
      * @return true when the Kafka cluster is exposed using load balancer.
      */
     public boolean isExposedWithLoadBalancer() {
-        return isExposed() && listeners.getExternal() instanceof KafkaListenerExternalLoadBalancer;
+        return !ListenersUtils.loadBalancerListeners(listeners).isEmpty();
```

Comment:
Better have helper methods on `ListenerUtils` which return whether there are any listeners of a given sort without having to allocate a `List` (that way you don't have to allocate a `List` only to test its emptiness).